cuda_core/build_hooks.py

```diff
@@ -54,6 +54,42 @@ def _get_proper_cuda_bindings_major_version() -> str:
     return "13"
 
 
+@functools.cache
+def _get_cuda_core_build_major_version() -> str:
+    # Explicit overwrite, e.g. in CI.
+    cuda_major = os.environ.get("CUDA_CORE_BUILD_MAJOR")
+    if cuda_major is not None:
+        return cuda_major
+
+    # Try to derive from the CUDA headers (preferred; matches what we compile against).
+    CUDA_PATH = os.environ.get("CUDA_PATH", os.environ.get("CUDA_HOME", None))
+    if CUDA_PATH:
+        for root in CUDA_PATH.split(os.pathsep):
+            cuda_h = os.path.join(root, "include", "cuda.h")
+            try:
+                with open(cuda_h, encoding="utf-8") as f:
+                    for line in f:
+                        m = re.match(r"^#\s*define\s+CUDA_VERSION\s+(\d+)\s*$", line)
+                        if m:
+                            v = int(m.group(1))
+                            # CUDA_VERSION is e.g. 12020 for 12.2.
+                            return str(v // 1000)
+            except OSError:
+                continue
+
+    # Fall back to driver-reported CUDA version if available.
+    try:
+        out = subprocess.run("nvidia-smi", env=os.environ, capture_output=True, check=True)  # noqa: S603, S607
+        m = re.search(r"CUDA Version:\s*([\d\.]+)", out.stdout.decode())
+        if m:
+            return m.group(1).split(".")[0]
+    except (FileNotFoundError, subprocess.CalledProcessError):
+        pass
```

Comment:
This is probably a bad idea because this is purely from the driver which is much more commonly installed separately from the toolkit and often will be significantly ahead of the toolkit.